lib/elixir/test/elixir/module/checker_test.exs

```diff
@@ -620,6 +620,188 @@ defmodule Module.CheckerTest do
     end
   end
 
+  describe "function header inference" do
+    test "warns on literals" do
```

Comment:
For completeness, we should find a way to write most of those tests without writing to files first.